lib/elixir/lib/system.ex

```diff
@@ -637,6 +637,26 @@ defmodule System do
     end
   end
 
+  @doc """
+  Similar to `System.cmd/3` but raises an error on non-zero exit status.
+
+  ## Examples
+
+      iex> System.cmd! "echo", ["hello"]
+      "hello\n"
+
+  """
+  @spec cmd!(binary, [binary], keyword) :: Collectable.t()
+  def cmd!(command, args, opts \\ []) when is_binary(command) and is_list(args) do
+    case cmd(command, args, opts) do
+      {result, 0} ->
+        result
+
+      {result, status} ->
+        raise "command exited with status #{inspect(status)}\n\n    Result: #{inspect(result)}"
```

Comment:
`inspect(result)` here was needed in case we collect results into something different than a string, e.g. `[]` etc. The downside is that if error is multiline it doesn't look great:

```
iex(16)> System.cmd!("elixir", ["-e", "1 + :a"], stderr_to_stdout: true)
** (RuntimeError) command exited with status 1

    Result: "** (ArithmeticError) bad argument in arithmetic expression\n    :erlang.+(1, :a)\n    (stdlib) erl_eval.erl:670: :erl_eval.do_apply/6\n    (elixir) lib/code.ex:192: Code.eval_string/3\n"
    (elixir) lib/system.ex:656: System.cmd!/3
```

maybe if collectable happens to be a binary, we pass it on without inspect so that exception gets `\n` and prints them?